src/main.cu

```diff
@@ -647,12 +655,12 @@ int main(int argc, const char** argv) {
                             csv_step << repetition << "," << gridWidth << "," << popSize << "," << pOccupation << "," << i << "," << step_time << "," << step_pop_count << std::endl;
                             // update average
                             sum_pop_count += step_pop_count;
+                            sum_step_time += step_time;
                         }
 
 
                         // log total simulation time
-                        const auto runTime = cudaSimulation.getElapsedTimeSimulation();
-                        const double averageStepTime = runTime / static_cast<double>(BENCHMARK_STEPS);
+                        const double averageStepTime = sum_step_time / static_cast<double>(BENCHMARK_STEPS);
```

Comment:
The average of the sum of individual times will be less accurate than using the total elapsed time, as it accumulates error within the timers.

Reply:
True but `cudaSimulation.getElapsedTimeSimulation()` return 0 if `step` is called rather than `simulate`

Reply:
In that case this question is really "why is step being used not simulate?". 

I.e we should be promoting the intended use of `simulate`, which calls init functions, exit functions and exit condition functions. The above use of step should be achievable with flamegpu features such as logging (or file-scoped std::vectors + a step function).